impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelProcessor.java

```diff
@@ -137,7 +139,7 @@ private Path doLocateExistingPom(Path project) {
         }
     }
 
-    private Model doRead(XmlReaderRequest request) throws IOException {
```

Comment:
this changes the whole code as this "can not happen"; or if needed then we should change API, if we can, to make code more concise.

<img width="1731" alt="image" src="https://github.com/user-attachments/assets/414af308-5c71-4c5c-ab45-c4d0bcae14c0" />